assets/js/blocks/stock-filter/block.tsx

```diff
@@ -50,14 +54,11 @@ const StockStatusFilterBlock = ( {
 		false
 	);
 
-	const [ hideOutOfStockItems ] = useState(
-		getSetting( 'hideOutOfStockItems', false )
-	);
-	const [ { outofstock, ...otherStockStatusOptions } ] = useState(
-		getSetting( 'stockStatusOptions', {} )
-	);
+	const [ { outofstock, ...otherStockStatusOptions } ] = useState<
+		Record< string, string >
+	>( getSetting( 'stockStatusOptions', {} ) );
```

Comment:
Not introduced in this PR, but I'm a bit confused by why we store this in the state. Couldn't this simply be?
```suggestion
	const { outofstock, ...otherStockStatusOptions } = getSetting(
		'stockStatusOptions',
		{}
	);
```

Reply:
Agreed!